src/orion/benchmark/task/task.py

```diff
@@ -0,0 +1,298 @@
+
+        Args:
+            hp (Union[HyperParameters, Dict, np.ndarray], optional):
+                Either a Hyperparameter dataclass, a dictionary, or a numpy
+                array containing the values for each dimension. Defaults to
+                None.
+
+        Returns:
+            np.ndarray: The performances of the hyperparameter on the sampled
+            task.
+        """
+
+        # return super().__call__(**kwargs)
+
+    def call(self, *args, **kwargs):
+        return self(*args, **kwargs)
+
+    @property
+    def max_trials(self) -> int:
+        """Return the max number of trials to run with this objective."""
+        return self._max_trials
+
+    def get_search_space(self) -> Dict[str, str]:
+        """Return the search space for the task objective function"""
+        return self.space
+
+    @property
+    def configuration(self):
+        """Return the configuration of the task."""
+        return {
+            self.__class__.__qualname__: {
+                "task_id": self.task_id,
+                "seed": self.seed,
+                "fixed_values": self.fixed_values,
+                "max_trials": self.max_trials,
+            }
+        }
+
+    @property
+    def full_space(self) -> Dict[str, str]:
+        """ Auto-generated orion-style 'space' dict for `self.hparams`.
+
+        This is generated using the `self.hparams` dataclass type.
+
+        Returns:
+            Dict[str, str]: Dict mapping from dimension name to the 'space'.
+        """
+        return self.hparams.get_orion_space_dict()
+
+    @classmethod
+    def get_domain(cls) -> Domain:
+        bounds = cls.hparams.get_bounds()
+        return Domain(
+            lower=np.asfarray([bound.domain[0] for bound in bounds]),
+            upper=np.asfarray([bound.domain[1] for bound in bounds]),
+        )
+
+    @property
+    def space(self) -> Dict[str, str]:
+        """ Auto-generated orion-style 'space' strings for `self.hparams`.
+
+        Entries from `self.fixed_values` are removed from the returned dictionary.
+        """
+        space = self.full_space.copy()
+        for k in self.fixed_values:
+            space.pop(k)
+        return space
+
+    @property
+    def full_domain(self) -> List[Dict[str, Any]]:
+        """ Auto-generated GPyOpt-style 'domain' dicts for `self.hparams`. """
+        return self.hparams.get_bounds_dicts()
+
+    @property
+    def domain(self) -> List[Dict[str, Any]]:
+        """ Auto-generated GPyOpt-style 'domain' dicts for `self.hparams`.
+
+        Entries with a name in `self.fixed_values` are not returned.
+        """
+        domains = self.full_domain.copy()
+        return list(filter(lambda d: d["name"] not in self.fixed_values, domains))
+
+    def seed_rng(self, seed: Optional[int] = None):
+        self.seed = seed
+        self.rng = np.random.default_rng(seed)
+        self.hparams.rng = self.rng
+
+    def sample(self, n: int = 1) -> List[HParams]:
```

Comment:
I think it's a bit weird that we sample from a Task. Maybe we should rather use a new space object representing the subspace (without fixed_values) instead, and then we could sample from the subspace directly.

Reply:
Removed those methods for now. I am considering adding a `space` property on the `BaseTask` class though, which would give back a `Space` object which I could then sample points from.